backend/src/main/java/de/bund/digitalservice/ris/norms/adapter/input/restapi/formatter/XmlFormattingResponseBodyAdvice.java

```
@@ -0,0 +1,80 @@
+import org.xml.sax.InputSource;
+
+/**
+ * ResponseBodyAdvice that formats XML responses
+ */
+@ControllerAdvice
+public class XmlFormattingResponseBodyAdvice implements ResponseBodyAdvice<Object> {
+
+  @Override
+  public boolean supports(
+    MethodParameter returnType,
+    Class<? extends HttpMessageConverter<?>> converterType
+  ) {
+    return true;
+  }
+
+  @Override
+  public Object beforeBodyWrite(
+    Object body,
+    MethodParameter returnType,
+    MediaType selectedContentType,
+    Class<? extends HttpMessageConverter<?>> selectedConverterType,
+    ServerHttpRequest request,
+    ServerHttpResponse response
+  ) {
+    if (
+      selectedContentType != null &&
+      selectedContentType.isCompatibleWith(MediaType.APPLICATION_XML) &&
+      body instanceof String xmlBody
+    ) {
+      return formatXml(xmlBody);
+    }
+    return body;
+  }
+
+  private String formatXml(String xml) {
+    try {
+      DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+      dbf.setNamespaceAware(true);
+      dbf.setFeature("http://apache.org/xml/features/disallow-doctype-decl", true);
+      dbf.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+      dbf.setExpandEntityReferences(false);
+      Document document = dbf.newDocumentBuilder().parse(new InputSource(new StringReader(xml)));
+
+      DOMImplementationLS impl = (DOMImplementationLS) document
+        .getImplementation()
+        .getFeature("LS", "3.0");
+      LSSerializer serializer = impl.createLSSerializer();
+
+      serializer.getDomConfig().setParameter("format-pretty-print", Boolean.TRUE);
+      serializer.getDomConfig().setParameter("xml-declaration", Boolean.FALSE);
+
+      LSOutput lsOutput = impl.createLSOutput();
+      StringWriter writer = new StringWriter();
+      lsOutput.setCharacterStream(writer);
+
+      serializer.write(document, lsOutput);
+      return writer.toString();
+    } catch (Exception e) {
+      return xml;
```

Please add logging in case of an exception, otherwise will always fail silently. For that add at the class level the annotation `@Slf4j` and then add a line with `log.error(`) and some descriptive message. Thanks!